src/main/java/net/earthcomputer/clientcommands/command/MapCommand.java

```diff
@@ -0,0 +1,90 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.block.MapColor;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.texture.NativeImage;
+import net.minecraft.client.util.ScreenshotRecorder;
+import net.minecraft.item.FilledMapItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.map.MapState;
+import net.minecraft.text.ClickEvent;
+import net.minecraft.text.Text;
+
+import java.io.File;
+import java.io.IOException;
+
+import static com.mojang.brigadier.arguments.IntegerArgumentType.getInteger;
+import static com.mojang.brigadier.arguments.IntegerArgumentType.integer;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+
+public class MapCommand {
+    private static final SimpleCommandExceptionType NO_HELD_MAP = new SimpleCommandExceptionType(Text.translatable("commands.cmap.noHeld"));
+
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cmap")
+            .then(literal("export")
+                .executes(ctx -> createMap(ctx.getSource(), 1))
+                .then(argument("upscale", integer(1, 16))
+                    .executes(ctx -> createMap(ctx.getSource(), getInteger(ctx, "upscale")))
+                )
+            )
+        );
+    }
+
+    private static int createMap(FabricClientCommandSource source, int upscale) throws CommandSyntaxException {
+        ClientPlayerEntity player = source.getPlayer();
+
+        ItemStack map;
+
+        // detect if map in hand
+        if (player.getMainHandStack().getItem() instanceof FilledMapItem) {
+            map = player.getMainHandStack();
+        } else if (player.getOffHandStack().getItem() instanceof FilledMapItem) {
+            map = player.getOffHandStack();
+        } else {
+            throw NO_HELD_MAP.create();
+        }
+
+        Integer mapId = FilledMapItem.getMapId(map);
+        MapState mapState = FilledMapItem.getMapState(mapId, player.world);
+        if (mapState == null) {
+            throw NO_HELD_MAP.create();
+        }
```

Comment:
Could this be abstracted to a separate method so that other subcommands could use it in the future?